src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt

```diff
@@ -78,10 +77,10 @@ class CargoWorkspace private constructor(
 
     private val targetByCrateRootUrl = packages.flatMap { it.targets }.associateBy { it.crateRootUrl }
 
-    fun findCrateByName(normName: String): Target? =
-        packages
-            .mapNotNull { it.libTarget }
-            .find { it.normName == normName }
+    fun findCrateByName(normName: String, inPackage: Package? = null): Target? =
```

Comment:
I totally agree on moving the method to `Package` and only leaving the "approximate" part in the workspace. But I have two concerns about the `Target`:

First, I don't see the benefits of having the method there. I understand that Rust itself tracks dependencies at the target level, but why should we do the same in IDE? If I'm not mistaken, one cannot have two different versions of the same crate in different targets, so it's safe to search in a common list of packages and separating them will only bring unnecessary complexity and potential problems. Or am I wrong? Is there an example where such separation is needed for correct resolving?

Second, at first glance, `cargo metadata` doesn't provide enough information to properly separate package dependencies among different targets. Resolve nodes have no information about targets, and values in the `packages.dependencies.target` field look sort of useless to me. But honestly, I didn't dig deep here yet.